source/extensions/filters/http/jwt_authn/filter_config.cc

```diff
@@ -22,6 +23,21 @@ FilterConfigImpl::FilterConfigImpl(
 
   jwks_cache_ = JwksCache::create(proto_config_, context, Common::JwksFetcher::create, stats_);
 
+  // Validate provider URIs.
+  // Note that the PGV well-known regex for URI is not implemented in C++, otherwise we could add a
+  // PGV rule instead of doing this check manually.
+  for (const auto& provider_pair : proto_config_.providers()) {
+    auto provider_value = std::get<1>(provider_pair);
+    if (provider_value.has_remote_jwks()) {
+      absl::string_view provider_uri = provider_value.remote_jwks().http_uri().uri();
+      Http::Utility::Url url;
+      if (!url.initialize(provider_uri, /*is_connect=*/false)) {
```

Comment:
it may be better to split `initialize()` to validation only (rather than actually allocating memory for the url), but this should be done in a future PR.

Reply:
+1. It will be great if we can build a light-weight validation utility function.  Overhead might be less of concern in config path, compared with data plane path. But we should optimize if possible.

Alternatively, we can look at PGV approach (not sure the effort there though)